### ISSUE TITLE
test_spmv_bsrmatrix_controls_pattern incorrect epsilon?

### ISSUE THREAD

**Issue by e10harvey** [kokkos/kokkos-kernels#1792]
In #1774, I noticed that the epsilon for this test is always using the float16 epsilon while it looks like this test runs for all types: float, double, etc. Shouldn't epsilon be:
```
static_cast<double>(
        std::numeric_limits<typename y_view_t::value_type>::epsilon());
```
Instead of 
```
static_cast<double>(
        std::numeric_limits<Kokkos::Experimental::half_t>::epsilon());
```
in `Test_sparse_spmv.hpp`?

**Comment by e10harvey** [comment]
Relevant discussion thread: https://github.com/kokkos/kokkos-kernels/pull/1774#discussion_r1165952472

**Comment by cwpearson** [comment]
The float16 tolerance should only be used when the algorithm under test is the tensor core algorithm. I wonder if this slipped through a refactor somehow.

**Comment by cwpearson** [comment]
This is resolved by removing the code in #1922
